smack-websocket/src/main/java/org/jivesoftware/smack/websocket/XmppWebsocketTransportModuleDescriptor.java

```diff
@@ -0,0 +1,125 @@
+/**
+ *
+ * Copyright 2020 Aditya Borikar
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.smack.websocket;
+
+import java.net.URI;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.jivesoftware.smack.c2s.ModularXmppClientToServerConnection;
+import org.jivesoftware.smack.c2s.ModularXmppClientToServerConnectionConfiguration;
+import org.jivesoftware.smack.c2s.ModularXmppClientToServerConnectionModule;
+import org.jivesoftware.smack.c2s.ModularXmppClientToServerConnectionModuleDescriptor;
+import org.jivesoftware.smack.c2s.internal.ModularXmppClientToServerConnectionInternal;
+import org.jivesoftware.smack.fsm.StateDescriptor;
+import org.jivesoftware.smack.websocket.XmppWebsocketTransportModule.EstablishingWebsocketConnectionStateDescriptor;
+
+/**
+ * The descriptor class for {@link XmppWebsocketTransportModule}.
+ * <br>
+ * To add {@link XmppWebsocketTransportModule} to {@link ModularXmppClientToServerConnection},
+ * use {@link ModularXmppClientToServerConnectionConfiguration.Builder#addModule(ModularXmppClientToServerConnectionModuleDescriptor)}.
+ */
+public final class XmppWebsocketTransportModuleDescriptor extends ModularXmppClientToServerConnectionModuleDescriptor {
+    private boolean performWebsocketEndpointDiscovery;
+    private URI uri;
+
+    public XmppWebsocketTransportModuleDescriptor(Builder builder) {
+        this.performWebsocketEndpointDiscovery = builder.performWebsocketEndpointDiscovery;
+        this.uri = builder.uri;
+    }
+
+    /**
+     * Returns true if websocket endpoint discovery is true, returns false otherwise.
+     * @return boolean
+     */
+    public boolean isWebsocketEndpointDiscoveryEnabled() {
+        return performWebsocketEndpointDiscovery;
+    }
+
+    /**
+     * Returns URI of the fallback websocket endpoint.
+     * @return uri
+     */
+    public URI getUri() {
```

Comment:
`getUri()` is not really meaningful, and the javadoc is not correct. This is not always a fallback websocket endpoint, isn't it?

I suggest to rename to `getExplicitlyProvidedUri()` (or something similar).